server/schedule/region_scatterer.go

```diff
@@ -250,18 +316,34 @@ func (r *RegionScatterer) collectAvailableStores(region *core.RegionInfo, contex
 	return targets
 }
 
-func (r *RegionScatterer) searchLeastleaderStore(peers map[uint64]*metapb.Peer, context engineContext) uint64 {
+// selectAvailableLeaderStores select the target leader store from the candidates. The candidates would be collected by
+// the existed peers store depended on the leader counts in the group level.
+func (r *RegionScatterer) selectAvailableLeaderStores(group string, peers map[uint64]*metapb.Peer, context engineContext) uint64 {
 	m := uint64(math.MaxUint64)
-	id := uint64(0)
+	tableLeaderStoreCandidates := map[uint64]struct{}{}
```

Comment:
Why need `tableLeaderStoreCandidates`? I think we can pick the min leader store directly.

Reply:
If there exist multipule stores have the same leader count(also the minest) in group level, we will choose the one which have the least leader count in cluster level.

Reply:
updated.